src/main/java/model/Money.java

```diff
@@ -0,0 +1,41 @@
+package model;
+
+import java.math.BigDecimal;
+
+import static model.Lotto.LOTTO_PRICE;
+
+public class Money {
+    private static final String MONEY_MUST_BE_POSITIVE = "구입금액은 0 이상의 숫자여야 합니다.";
+    private static final int IS_NEGATIVE = -1;
+
+    private final BigDecimal amount;
+
+    public Money(BigDecimal amount) {
+        validateAmount(amount);
+        this.amount = amount;
+    }
+
+    private void validateAmount(BigDecimal amount) {
+        int status = amount.compareTo(BigDecimal.ZERO);
+        if (status == IS_NEGATIVE) {
+            throw new IllegalArgumentException(MONEY_MUST_BE_POSITIVE);
+        }
+    }
+
+    public int calculateIssueCount() {
+        return amount.divide(LOTTO_PRICE).intValue();
+    }
+
+    public BigDecimal calculateTotalSpendMoney() {
+        return amount.subtract(amount.remainder(LOTTO_PRICE));
+    }
+
+    public BigDecimal multiplyCount(int winningCount) {
+        return amount.multiply(new BigDecimal(winningCount));
+    }
```

Comment:
Money 객체를 불변객체로 만들고 싶은 의도로 보여요 : ) 
지금 Money의 연산 기능들의 반환 타입은 자기 자신과 같은 타입이어야 한다고 생각해요. 
또한 Money는 VO 객체로 볼 수 있을 것 같네요 : ) 
VO 객체에 대한 피드백은 이전에 드렸기 때문에 Money 객체가 VO 객체로 만드려면 어떻게 처리할 수 있을지에 대해 고민해볼 수 있을 것 같아요. 

지금은 변경해야 할 것들이 너무 많아지기 때문에 변경하지 않아도 괜찮지만 

[불변객체를 만드는 방법](https://woowacourse.github.io/javable/post/2020-05-18-immutable-object/) <br> 
JPA를 사용해본적이 있으시니(~~Entity가 꼭 JPA만 해당하는 개념은 아니지만..~~) 해당 글을 읽어보는 것도 좋을 것같아요! [DTO vs VO vs Entity](https://jordy-torvalds.tistory.com/81)  


```suggestion
    public Money calculateTotalSpendMoney() {
        BigDecimal spendMoney = amount.remainder(LOTTO_PRICE);
        return new Money(amount.subtract(spendMoney));
    }

    public Money multiplyCount(int winningCount) {
        BigDecimal winningCountNumber = new BigDecimal(winningCount);
        return new Money(amount.multiply(winningCountNumber));
    }
```

Reply:
Money 객체가 연산을 했을 때 반환 타입이 Money여야 한다는 의미가 이해가 잘 되지 않는 것 같습니다. 고민해보는 시간을 가져보도록 하겠습니다. 감사합니다.